packages/api/src/apps/datastore_api/app.ts

```diff
@@ -0,0 +1,327 @@
+  const web3 = getWeb3(env.CUSTOM_NODE_URL);
+
+  // how often to run expensive state updates, defaults to 10 minutes
+  const updateRateS = Number(env.UPDATE_RATE_S || 10 * 60);
+  // Defaults to 60 seconds, since this is i think a pretty cheap call, and we want to see new contracts quickly
+  const detectContractsUpdateRateS = Number(env.DETECT_CONTRACTS_UPDATE_RATE_S || 60);
+  // Defaults to 15 minutes, prices dont update in coingecko or other calls very fast
+  const priceUpdateRateS = Number(env.PRICE_UPDATE_RATE_S || 15 * 60);
+  // default to 10 days worth of blocks
+  const oldestBlock = Number(env.OLDEST_BLOCK_MS || 10 * 60 * 60 * 24 * 1000);
+
+  assert(updateRateS >= 1, "UPDATE_RATE_S must be 1 or higher");
+  assert(detectContractsUpdateRateS >= 1, "DETECT_CONTRACTS_UPDATE_RATE_S must be 1 or higher");
+  assert(priceUpdateRateS >= 1, "PRICE_UPDATE_RATE_S must be 1 or higher");
+
+  // services can emit events when necessary, though for now any services that depend on events must be in same process
+  const serviceEvents = new Events();
+  const { datastores } = storesFactory;
+  // state shared between services
+  const appState: AppState = {
+    provider,
+    web3,
+    coingecko: new Coingecko(),
+    zrx: new Zrx(env.zrxBaseUrl),
+    blocks: tables.blocks.Table("Block", datastores.blockStore),
+    emps: {
+      active: tables.emps.Table("Active Emp", datastores.empsActiveStore),
+      expired: tables.emps.Table("Expired Emp", datastores.empsExpiredStore),
+    },
+    prices: {
+      usd: {
+        latest: {},
+        history: {},
+      },
+    },
+    synthPrices: {
+      latest: {},
+      history: {},
+    },
+    marketPrices: {
+      usdc: {
+        latest: {},
+        history: empStatsHistory.Table("Market Price", datastores.empStatsHistoryStore),
+      },
+    },
+    erc20s: tables.erc20s.Table("Erc20", datastores.erc20Store),
+    stats: {
+      emp: {
+        usd: {
+          latest: {
+            tvm: empStats.Table("Latest Tvm", datastores.empStatsTvmStore),
+            tvl: empStats.Table("Latest Tvl", datastores.empStatsTvlStore),
+          },
+          history: {
+            tvm: empStatsHistory.Table("Tvm History", datastores.empStatsTvlHistoryStore),
+            tvl: empStatsHistory.Table("Tvl History", datastores.empStatsTvmHistoryStore),
+          },
+        },
+      },
+      lsp: {
+        usd: {
+          latest: {
+            tvl: empStats.Table("Latest Tvl", datastores.lspStatsTvlStore),
+            tvm: empStats.Table("Latest Tvm", datastores.lspStatsTvmStore),
+          },
+          history: {
+            tvl: empStatsHistory.Table("Tvl History", datastores.lspStatsTvlHistoryStore),
+          },
+        },
+      },
+      global: {
+        usd: {
+          latest: {
+            tvl: [0, "0"],
+          },
+          history: {
+            tvl: empStatsHistory.Table("Tvl Global History"),
+          },
+        },
+      },
+    },
+    lastBlockUpdate: 0,
```

Comment:
we can make a separate issue for this, but the lastblockupdate should default to the latest block processed and persisted in store, or fallback to 0 if none was saved